test/philomena_web/controllers/reactivation_controller_test.exs

```diff
@@ -0,0 +1,53 @@
+defmodule PhilomenaWeb.ReactivationControllerTest do
+  use PhilomenaWeb.ConnCase, async: true
+  import Philomena.TestUtilities
+
+  alias Swoosh.Adapters.Local.Storage.Memory
+  alias Philomena.Users
+
+  setup :register_and_log_in_user
+
+  @host PhilomenaWeb.Endpoint.config(:url)[:host]
+  @port PhilomenaWeb.Endpoint.config(:http)[:port]
+
+  describe "GET /reactivations/:id" do
+    test "renders the reactivate account page", %{conn: conn} do
+      conn = delete(conn, ~p"/deactivations")
+      conn = get(conn, ~p"/reactivations/pinkie-pie-is-best-pony")
+      response = html_response(conn, 200)
+      assert response =~ "<h1>Reactivate Your Account</h1>"
+    end
+  end
+
+  describe "POST /reactivations/" do
+    test "reactivate account page works", %{conn: conn, user: user} do
+      conn = delete(conn, ~p"/deactivations")
+
+      {token, url} =
```

Comment:
Instead of extracting from email, it should be possible to do this in the way UnlockControllerTest does it:
```elixir
      token =
        extract_user_token(fn url ->
          Users.deliver_user_reactivation_instructions(user, url)
        end)
```

Checking the email may be introducing nondeterminism since email is sent on a background job.